src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java

```diff
@@ -127,16 +138,56 @@ public List<LinkedFile> findAssociatedNotLinkedFiles(BibEntry entry) throws IOEx
 
             if (!fileAlreadyLinked) {
                 Optional<ExternalFileType> type = FileUtil.getFileExtension(foundFile)
-                                                            .map(extension -> ExternalFileTypes.getExternalFileTypeByExt(extension, filePreferences))
-                                                            .orElse(Optional.of(new UnknownExternalFileType("")));
+                                                          .map(extension -> ExternalFileTypes.getExternalFileTypeByExt(extension, filePreferences))
+                                                          .orElse(Optional.of(new UnknownExternalFileType("")));
 
                 String strType = type.isPresent() ? type.get().getName() : "";
                 Path relativeFilePath = FileUtil.relativize(foundFile, directories);
                 LinkedFile linkedFile = new LinkedFile("", relativeFilePath, strType);
                 linkedFiles.add(linkedFile);
             }
         }
-
         return linkedFiles;
     }
+
+    public RelinkedResults relinkingFiles(List<LinkedFile> listlinked) {
+        List<LinkedFile> changedFiles = new ArrayList<>();
+        List<IOException> exceptions = new ArrayList<>();
+
+        for (LinkedFile file : listlinked) {
+            Path path = Path.of(file.getLink());
+            if (!Files.exists(path)) {
+                Path filePath = Path.of(file.getLink());
+                String directoryPath = filePath.getParent().getParent().toString();
+
+                String fileNameString = filePath.getFileName().toString();
+
+                List<String> fileLocations = searchFileInDirectoryAndSubdirectories(Path.of(directoryPath), fileNameString);
+                if (!fileLocations.isEmpty()) {
+                    file.setLink(fileLocations.get(0));
+                    changedFiles.add(file);
+                } else {
+                    exceptions.add(new IOException("couldn't find file: " + file.getLink()));
+                }
+            }
+        }
+        return new RelinkedResults(changedFiles, exceptions);
```

Comment:
You collect filenot found only - why not just listing the filenames which are not found in a list?